tests/tools/test_parmetis_wrapper.py

```diff
@@ -0,0 +1,271 @@
+            ['author:writes:paper', 'year'],
+            ['paper:rev_writes:author', 'year']
+        ]:
+            output_edge_sub_dir = os.path.join(output_edge_data_dir, etype)
+            for i in range(num_chunks):
+                chunk_f_name = '{}-{}.npy'.format(feat, i)
+                chunk_f_name = os.path.join(output_edge_sub_dir, chunk_f_name)
+                assert os.path.isfile(chunk_f_name)
+                feat_array = np.load(chunk_f_name)
+                assert feat_array.shape[0] == num_edges[etype] // num_chunks
+
+        # Trigger ParMETIS wrapper here. 
+        schema_path = os.path.join(root_dir, 'chunked-data/metadata.json')
+        results_dir = os.path.join(root_dir, 'parmetis-data')
+        os.system('mpirun -np 2 python3 tools/distpartitioning/parmetis_wrapper.py '\
+                  '--schema {} --output {}'.format(schema_path, results_dir))
+
+        # Now add all the tests and check whether the test has passed or failed
+        # Test - 1
+        # Read parmetis_nfiles and ensure all files are present
+        parmetis_data_dir = os.path.join(root_dir, 'parmetis-data')
+        assert os.path.isdir(parmetis_data_dir)
+        parmetis_nodes_file = os.path.join(parmetis_data_dir, 'parmetis_nfiles.txt')
+        assert os.path.isfile(parmetis_nodes_file)
+
+        with open(parmetis_nodes_file, 'r') as f:
+            lines = f.readlines()
+            total_node_count = 0
+            for line in lines:
+                tokens = line.split(" ")
+                print(tokens)
+                assert len(tokens) == 3
+                print(tokens[0])
+                assert os.path.isfile(tokens[0])
+                assert int(tokens[1]) == total_node_count
+
+                #check contents of each of the nodes files here
+                with open(tokens[0], 'r') as nf:
+                    node_lines = nf.readlines()
+                    node_count = len(node_lines)
+                    total_node_count += node_count
+                assert int(tokens[2]) == total_node_count
+        
+        # Count the total no. of nodes
+        true_node_count = 0
+        num_nodes_per_chunk = meta_data['num_nodes_per_chunk']
+        for i in range(len(num_nodes_per_chunk)):
+            node_per_part = num_nodes_per_chunk[i]
+            for j in range(len(node_per_part)):
+                true_node_count += node_per_part[j]
+        assert total_node_count == true_node_count
+
+        # Read parmetis_efiles and ensure all files are present
+        parmetis_edges_file = os.path.join(parmetis_data_dir, 'parmetis_efiles.txt')
+        assert os.path.isfile(parmetis_edges_file)
+
+        with open(parmetis_edges_file, 'r') as f:
+            lines = f.readlines()
+            total_edge_count = 0
+            for line in lines:
+                ff = line.strip()
+                assert os.path.isfile(ff)
+
+                with open(ff, 'r') as ef:
+                    edge_count = len(ef.readlines())
```

Comment:
Please note that for the edges each line will have following format: <global_src_id> <global_dst_id> 
Others are omitted, since ParMETIS does not care about the rest of the line.

Reply:
Again the processing overhead will be too much and will make the CI process very time consuming. I prefer it this way.

Reply:
same here, added the logic here you requested. if this creates issues later we can comment it out.